File: src/main/java/net/vulcanmc/vulcaneconomy/commands/Transactions.java
package net.vulcanmc.vulcaneconomy.commands;

import net.vulcanmc.vulcaneconomy.VulcanEconomy;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

public class Transactions implements CommandExecutor {
    private final VulcanEconomy plugin;

    public Transactions(VulcanEconomy plugin) {
        this.plugin = plugin; // Store the plugin in situations where you need it.
    }

    @Override
    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
        if(args.length == 1) {
             sender.sendMessage(ChatColor.GOLD + "===== {PLAYERS}'s Transactions =====");
             sender.sendMessage(ChatColor.AQUA + "7 | DEC23 11:59 | D-$500 | Payment to That1Guy2");
             sender.sendMessage(ChatColor.AQUA + "43 | DEC24 1:54 | D-$530 | Payment to That1Guy2");
             sender.sendMessage(ChatColor.AQUA + "198 | DEC27 16:23 | C-$53 | Payment from Notch");
             sender.sendMessage(ChatColor.AQUA + "44885 | DEC29 17:42 | D-$56 | Payment to Notch");
            sender.sendMessage(ChatColor.AQUA + "425893 | DEC31 5:34 | C-$993 | Payment from puppy3276");
             sender.sendMessage(ChatColor.GOLD + "Page 1 out of 5");
            return true;
        }
        return false;
    }
}

File: src/main/java/net/vulcanmc/vulcaneconomy/commands/Transactions.kt
package net.vulcanmc.vulcaneconomy.commands

import net.vulcanmc.vulcaneconomy.VulcanEconomy
import net.vulcanmc.vulcaneconomy.rest.Account
import net.vulcanmc.vulcaneconomy.rest.Currency
import net.vulcanmc.vulcaneconomy.rest.Transaction
import net.vulcanmc.vulcaneconomy.rest.User
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.OfflinePlayer
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.time.format.DateTimeFormatter

import java.util.ArrayList

class Transactions(private val plugin: VulcanEconomy)// Store the plugin in situations where you need it.
    : CommandExecutor {

    fun isNumeric(strNum: String): Boolean {
        try {
            val d = java.lang.Double.parseDouble(strNum)
        } catch (nfe: NumberFormatException) {
            return false
        } catch (nfe: NullPointerException) {
            return false
        }
        return true
    }

    override fun onCommand(sender: CommandSender, cmd: Command, label: String, args: Array<String>): Boolean {
        var player: OfflinePlayer
        var quantity : Int = 10
        var currencyPos : Int = 1
        if (args.size >= 1) {
            if(isNumeric(args[0]) && sender is Player) {
              player=sender;
              quantity = Integer.valueOf(args[0]);
            } else {
                player= Bukkit.getOfflinePlayer(args[0]);
            }
        } else if(sender is Player) {
            player = sender;
        } else {
            return false;
        }
        var currency : Currency = plugin.currencies.defaultCurrency;
        if(args.size >=2) {
            var key = args[1];
            if(isNumeric(key)) {
                quantity = Integer.valueOf(key)
                currencyPos = 2;
            }
            if(args.size >= currencyPos+1) {
                key = args[currencyPos]
                if(plugin.currencies.getCurrency(key) != null) {
                    currency = plugin.currencies.getCurrency(key)!!;
                } else {
                    sender.sendMessage(plugin.prefix + ChatColor.RED + "Invalid currency. Type /eco currencies")
                    return false;
                }
            }
        }

            sender.sendMessage(ChatColor.GOLD.toString() + "===== " + player.name + "'s Transactions (${currency.name}) =====")
        val inputFormat = DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm:ss")
        val user = User(player.uniqueId)
        val account = user.getAccount(currency)
        val transactions = account!!.getTransactions(quantity)
        plugin.logger.info(transactions.size.toString())

        for (transaction in transactions) {
            //println(account.id.toString() + transaction.creditPlayer?.id.toString())
            if(account.id.equals(transaction.creditPlayer?.id)) {
                sender.sendMessage(ChatColor.RED.toString() + "$${transaction.amount.toString()} | ${transaction.time} | ${transaction.plugin} | ${transaction.description}")
            } else {
                sender.sendMessage(ChatColor.GREEN.toString() + "$${transaction.amount.toString()} | ${transaction.time} | ${transaction.plugin} | ${transaction.description}")
            }
        }
            return true
    }
}
